bottlecap/src/traces/trace_processor.rs

```diff
@@ -0,0 +1,329 @@
+// Copyright 2023-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::tags::provider;
+use datadog_trace_obfuscation::obfuscation_config;
+use datadog_trace_utils::config_utils::trace_intake_url;
+use datadog_trace_utils::tracer_payload::TraceEncoding;
+use ddcommon::Endpoint;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use hyper::{http, Body, Request, Response, StatusCode};
+use tokio::sync::mpsc::Sender;
+use tracing::info;
+
+use crate::config;
+use crate::traces::http_utils::{self, log_and_create_http_response};
+use datadog_trace_obfuscation::obfuscate::obfuscate_span;
+use datadog_trace_utils::trace_utils::SendData;
+use datadog_trace_utils::trace_utils::{self};
+
+use super::trace_agent::{ApiVersion, MAX_CONTENT_LENGTH};
+
+#[async_trait]
+pub trait TraceProcessor {
+    /// Deserializes traces from a hyper request body and sends them through the provided tokio mpsc
+    /// Sender.
+    async fn process_traces(
+        &self,
+        config: Arc<config::Config>,
+        req: Request<Body>,
+        tx: Sender<trace_utils::SendData>,
+        tags_provider: Arc<provider::Provider>,
+        version: ApiVersion,
+    ) -> http::Result<Response<Body>>;
+}
+#[derive(Clone)]
+#[allow(clippy::module_name_repetitions)]
+pub struct ServerlessTraceProcessor {
+    pub obfuscation_config: Arc<obfuscation_config::ObfuscationConfig>,
+}
+
+#[async_trait]
+impl TraceProcessor for ServerlessTraceProcessor {
+    async fn process_traces(
+        &self,
+        config: Arc<config::Config>,
+        req: Request<Body>,
+        tx: Sender<trace_utils::SendData>,
+        tags_provider: Arc<provider::Provider>,
+        version: ApiVersion,
+    ) -> http::Result<Response<Body>> {
+        info!("Recieved traces to process");
```

Comment:
```suggestion
        debug!("Received traces to process");
```

Maybe make it a debug? Might need an import